test/test_torch.py

```diff
@@ -3903,6 +3910,13 @@ def test_print(self):
             self.assertEqual(x.__repr__(), str(x))
             self.assertExpectedInline(str(x), '''tensor([4.0000,    inf, 1.5000,   -inf, 0.0000,    nan, 1.0000])''')
 
+            y = torch.tensor([4, inf, complex(1.5, inf), complex(-inf, 4), 0, complex(nan, inf), 1])
```

Comment:
@mruberry here's the test for `NaN` and `inf` cases

Reply:
Can we add a case with nan in the imaginary part?

Reply:
sure added!